### PR TITLE
docs: OpenSource101 demo

### PR DESCRIPTION
**Pull request by satvik-s**
### SUMMARY

Add OSS101 examples/demo modules

### DETAILS

- add development module
- add test module

### CHECKLIST
- [ ] Documentation updated (if needed)
- [ ] Unit tests exist to cover the code you are changing and validated
- [ ] Integration tests exist to cover the code you are changing and validated

### CODE REVIEW

File: examples/open-source-101/test/development.spec.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    test('should respond back with successful response when spotify API is working', async () => {
+        // register successful Spotify API response
+        await mockServer.register(
+            // incoming request schema
+            {
+                method: 'GET',
+                endpoint: '/v1/artists/0TnOYISbd1XYRBk9myaseg',
+            },
+            // error response schema
```
Comment (catnguyen-hbo) [inline examples/open-source-101/test/development.spec.ts:23]: this is the happy path right? Not error response schema

Comment (satvik-s) [reply examples/open-source-101/test/development.spec.ts:23]: whooops

File: examples/open-source-101/test/development.spec.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        const resp = await app.get('/artist-popularity');
+        expect(resp.statusCode).toEqual(200);
+        expect(resp.body).toBeDefined();
+        expect(resp.body.artistName).toBeDefined();
+        expect(resp.body.artistPopularity).toBeDefined();
+        expect(resp.body.artistGenres).toBeDefined();
+        expect(resp.body.artistGenres.length).toBeGreaterThan(0);
+    });
+
+    // why use wiremock-captain
```
Comment (catnguyen-hbo) [inline examples/open-source-101/test/development.spec.ts:39]: did you mean to leave this comment here? What I mentioned earlier btw was just making it clear that there are two different use cases that people can incorporate with the mocks: 

1. Mocking a successful happy path that returns expected response 
2. Mocking a failure path that returns 401 

Like maybe emphasize(either when you're presenting or in a comment) that it's simple to change the response and status code and perform different scenarios. Like the 2 cases are basically identical except for a few changes.

**Review by catnguyen-hbo — DISMISSED** [review]

**Review by kbang-hbo — DISMISSED** [review]

File: examples/open-source-101/test/development.spec.ts (hunk 2 above)
Comment (satvik-s) [reply examples/open-source-101/test/development.spec.ts]: makes sense. thanks for the feedback!

**Review by catnguyen-hbo — APPROVED** [review]

File: examples/open-source-101/src/development-final.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        },
+    );
+
+    httpLogger.logger.info('started mock spotify server');
+}
+
+const app = express();
+app.use(httpLogger);
+
+// set up API
```
Comment (EmilyHolmez) [inline examples/open-source-101/src/development-final.ts:38]: I love that we're using the Spotify API for the demo - this helps demonstrate the power of the tool with a service most people are already familiar with! 

Love this choice of API for the demo.

**Review by EmilyHolmez — APPROVED** [review]

**Pull request merged**